python/cog/server/helpers.py

```diff
@@ -148,3 +159,37 @@ def run(self) -> None:
                     if drain_tokens_seen >= drain_tokens_needed:
                         self.drain_event.set()
                         drain_tokens_seen = 0
+
+X = TypeVar("X")
+
+# functionally this is the exact same thing as aioprocessing but 0.1% the code
+# however it's still worse than just using actual asynchronous io
+class AsyncPipe(Generic[X]):
+    def __init__(self, conn: Connection) -> None:
+        self.conn = conn
+        self.exiting = threading.Event()
+        self.executor = concurrent.futures.ThreadPoolExecutor(1)
+
+    def send(self, obj: Any) -> None:
+        self.conn.send(obj)
+
+    def shutdown(self) -> None:
+        self.exiting.set()
+        self.executor.shutdown(wait=False)
+        # if we ever need cancel_futures (introduced 3.9), we can copy it in from
+        # https://github.com/python/cpython/blob/3.11/Lib/concurrent/futures/thread.py#L216-L235
+
+    def poll(self, timeout: float = 0.0) -> bool:
+        return self.conn.poll(timeout)
+
+    def _recv(self) -> Optional[X]:
+        # this ugly mess could easily be avoided with loop.connect_read_pipe
+        # even loop.add_reader would help but we don't want to mess with a thread-local loop
+        while not self.exiting.is_set():
+            if self.conn.poll(0.01):
+                return self.conn.recv()
```

Comment:
Add an explicit `return` statement here to placate type checker

```suggestion
    def _recv(self) -> Optional[X]:
        # this ugly mess could easily be avoided with loop.connect_read_pipe
        # even loop.add_reader would help but we don't want to mess with a thread-local loop
        while not self.exiting.is_set():
            if self.conn.poll(0.01):
                return self.conn.recv()
        
        return None
```